### PR TITLE
Practice bot fixes / OI joystick remap

### PR DESCRIPTION
**Pull request by aschokking**
While testing on the practice bot fixed some bugs and smoothed out the operator joystick for arm control.

@JohnGilb

### CODE REVIEW

**Comment by aschokking** [comment]
I need to do a little cleanup on the core changes here and then merge them into master separately

File: CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java [hunk 1]
```diff
@@ -26,7 +26,24 @@ public void initialize() {
 
     @Override
     public void execute() {
-        armSubsystem.setArmMotorPower(-oi.operatorJoystick.getVector().y/2);
+        double desire = -oi.operatorJoystick.getVector().y;
```
Comment (WasabiFan) [inline CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java]: "desire" doesn't help a reader understand what the variable does. This should be changed to something more specific.

File: CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java [hunk 2]
```diff
@@ -26,7 +26,24 @@ public void initialize() {
...
+        // offset by range joystick kicks in
+        if(desire > 0) {
```
Comment (WasabiFan) [inline CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java]: Won't this make the arm move in the opposite direction if the joystick value is within 0.15 of 0?

File: CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java [hunk 3]
```diff
@@ -26,7 +26,24 @@ public void initialize() {
...
+            desire -= 0.15;
+        } else {
+            desire += 0.15;
+        }
+        
+        // rescale back to -1 to 1 range
+        desire = desire / 0.85;
+
```
Comment (WasabiFan) [inline CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java]: These statements should be cleaned up using `*=` and `/=`.

File: CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java [hunk 4]
```diff
@@ -26,7 +26,24 @@ public void initialize() {
...
+        }
+        
+        // rescale back to -1 to 1 range
+        desire = desire / 0.85;
+        
+        // square input
+        desire = desire * Math.abs(desire);
+        
+        // and scale down
+        desire = desire / 1.5;
```
Comment (WasabiFan) [inline CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java]: It is hard to mentally understand what dividing by a decimal greater than 1 does at first glance. Use multiplication here to make it more easily understood as scaling.

File: CompetitionTemplate/src/competition/subsystems/lighting/LightingSubsystem.java [hunk 5]
```diff
@@ -3,17 +3,19 @@
...
 import xbot.common.injection.wpi_factories.WPIFactory;
 
+@Singleton
 public class LightingSubsystem extends BaseSubsystem {
 
     private static Logger log = Logger.getLogger(LightingSubsystem.class);
     
     public enum LightingState {
-        Disabled(0), Enabled(1), BallCaptured(2);
+        Disabled(0), Enabled(2), BallCaptured(1);
```
Comment (WasabiFan) [inline CompetitionTemplate/src/competition/subsystems/lighting/LightingSubsystem.java]: These should be in order of number.

File: CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java (hunk 2 above)
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmManualControlCommand.java]: Right now the joystick value is never between -0.15 and 0.15

**Pull request merged**